### PR TITLE
[17.0][OU-ADD] onboarding: migrate to 17.0

### PR DESCRIPTION
**Pull request by daiduongnguyen-odoo**

### CODE REVIEW

File: openupgrade_scripts/scripts/onboarding/17.0.1.2/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def _onboarding_step_convert_field_m2o_to_m2m(env):
+    # Convert m2o to m2m in 'onboarding.onboarding.step'
+    openupgrade.m2o_to_x2m(
+        env.cr,
+        env["onboarding.onboarding.step"],
+        "onboarding_onboarding_step",
+        "onboarding_ids",
+        "onboarding_id",
```
Comment (remi-filament) [inline openupgrade_scripts/scripts/onboarding/17.0.1.2/post-migration.py:13]: onboarding_id need to be copied first in pre-commit as per https://oca.github.io/openupgradelib/API.html#openupgradelib.openupgrade.m2o_to_x2m

File: openupgrade_scripts/scripts/onboarding/17.0.1.2/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+def _onboarding_progress_step_convert_field_m2o_to_m2m(env):
+    """
+    Convert m2o to m2m in 'onboarding.progress.step'
+    """
+    openupgrade.m2o_to_x2m(
+        env.cr,
+        env["onboarding.progress.step"],
+        "onboarding_progress_step",
+        "progress_ids",
+        "progress_id",
```
Comment (remi-filament) [inline openupgrade_scripts/scripts/onboarding/17.0.1.2/post-migration.py:26]: progress_id need to be copied first in pre-commit as per https://oca.github.io/openupgradelib/API.html#openupgradelib.openupgrade.m2o_to_x2m

**Review by remi-filament — CHANGES_REQUESTED** [review]
Thanks @duong77476-viindoo I note here my review findings, I know you are not working on these ones anymore, it will serve in a second time to take over these PR !

File: openupgrade_scripts/scripts/onboarding/17.0.1.2/post-migration.py (hunk 1 above)
Comment (hbrunn) [reply openupgrade_scripts/scripts/onboarding/17.0.1.2/post-migration.py:13]: @remi-filament thanks for your input, but the recommendation applies only to very old versions of OpenUpgrade.

I filed a [PR](https://github.com/OCA/openupgradelib/pull/389) on openupgradelib to clarify this.

**Review by hbrunn — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**